aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -426,6 +433,99 @@ std::tuple<Tensor&,Tensor&> solve_out(Tensor& solution, Tensor& lu, const Tensor
   return std::tuple<Tensor&, Tensor&>(solution, lu);
 }
 
+
+// This is a type dispatching helper function for 'apply_solve'
+Tensor& _linalg_solve_out_helper_cpu(Tensor& result, Tensor& input, Tensor& infos) {
+  // 'result' and 'input' should be in column major order (it should be checked before calling this function)
+  // the content of 'result', 'input' and 'infos' is overriden by 'apply_solve'
+  // 'result' should contain data of 'other' tensor (right-hand-side of the linear system of equations)
+  // 'input' should contain data of original 'input' tensor (left-hand-side of the linear system of equations)
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(result.scalar_type(), "linalg_solve_out_cpu", [&]{
+    apply_solve<scalar_t>(result, input, infos);
+  });
+  return result;
+}
+
+// Solves a system of linear equations matmul(input, x) = other in-place
+// LAPACK/MAGMA error codes are saved in 'infos' tensor, they are not checked here
+static Tensor& linalg_solve_out_info(Tensor& result, Tensor& infos, const Tensor& input, const Tensor& other) {
+  TORCH_CHECK(infos.scalar_type() == kInt,
```

Comment:
Ack! My mistake, @IvanYashchuk. I forgot to ask about validating that the tensors are on the correct device. Is there a check that input and other are both on the same device?

Reply:
It is checked by calling [`_linalg_broadcast_batch_dims`](https://github.com/pytorch/pytorch/pull/48456/files#diff-76df047bca5c2d7a540bdcc4023d68641c4f15638d83c5c874d833b03f9636b9R473) which includes a call to a function `linearSolveCheckInputs` that does this check https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/LinearAlgebraUtils.h#L65-L68

Reply:
Thanks for verifying.